src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -906,10 +906,28 @@ private bool IsPossibleAttributeDeclaration()
 
         private SyntaxList<AttributeListSyntax> ParseAttributeDeclarations(bool inExpressionContext)
         {
-            var attributes = _pool.Allocate<AttributeListSyntax>();
             var saveTerm = _termState;
             _termState |= TerminatorState.IsAttributeDeclarationTerminator;
 
+            // An attribute can never appear *inside* an attribute declaration in the language.  However, during parsing
```

Comment:
>An attribute can never appear *inside* an attribute declaration in the language. [](http://example.com/codeflow?start=15&length=80)

It looks like the grammar allows an attribute argument expression to contain a lambda expression with an attribute on the lambda. This is only disallowed in binding because the lambda expression does not represent a constant, correct?

Reply:
> This is only disallowed in binding because the lambda expression does not represent a constant, correct?

Correct.  This is never going to be legal code.  So the decision here is to prevent it at parsing time vs binding time (and only for the attributed lambda case).  That way we can avoid pathlogical parsing conditions.

Pros: Pathological parsing cases don't go off hte rails in speculative parsing.
Cons: A syntactically legal (but semantically illegal) case errors earlier and produces a worse parse tree.

This seems like an easy justification to make.  The impact of speculative parsing here is enormous (it literally hangs vs/compiler).  While the impact of disallowing an attributed-lambda *in* an attribute at parsing time is extremely minor.  Who is realistically writing that?  And, even if they do, getting a different syntactic error is fine.